jaseci_core/jaseci/svc/common.py

```diff
@@ -125,12 +134,252 @@ def __init__(self):
+                self.check(self.namespace, svc, hook)
+            except Exception as e:
+                logger.exception(
+                    f"Error checking {svc} !\n" f"{e.__class__.__name__}: {e}"
+                )
+
+    ###################################################
+    #                   KUBERNETES                    #
+    ###################################################
+
+    def create(self, kind: str, name: str, namespace: str, conf: dict):
+        try:
+            logger.info(f"Creating {kind} for `{name}` with namespace `{namespace}`")
+            self.kubernetes.create(kind, namespace, conf)
+        except ApiException:
+            logger.error(
+                f"Error creating {kind} for `{name}` with namespace `{namespace}`"
+            )
+
+    def patch(self, kind: str, name: str, namespace: str, conf: dict):
+        try:
+            logger.info(f"Patching {kind} for `{name}` with namespace `{namespace}`")
+            self.kubernetes.patch(kind, name, namespace, conf)
+        except ApiException:
+            logger.error(
+                f"Error patching {kind} for `{name}` with namespace `{namespace}`"
+            )
+
+    def read(self, kind: str, name: str, namespace: str):
+        try:
+            logger.info(f"Retrieving {kind} for `{name}` with namespace `{namespace}`")
+            return self.kubernetes.read(kind, name, namespace=namespace)
+        except ApiException as e:
+            logger.error(
+                f"Error retrieving {kind} for `{name}` with namespace `{namespace}`"
+            )
+            return e
+
+    def delete(self, kind: str, name: str, namespace: str):
+        try:
+            logger.info(f"Deleting {kind} for `{name}` with namespace `{namespace}`")
+            return self.kubernetes.delete(kind, name, namespace=namespace)
+        except ApiException as e:
+            logger.error(
+                f"Error deleting {kind} for `{name}` with namespace `{namespace}`"
+            )
+            return e
+
+    def check(self, namespace, svc_name, hook):
+
+        svc = self.meta.get_service(svc_name, hook)
+
+        if not svc.is_running():
+            if svc.kube:
+                config_map = svc.kube
+                pod_name = ""
+                old_config_map = deepcopy(svc.old_kube)
+                for kind, confs in config_map.items():
+                    for conf in confs:
+                        name = conf["metadata"]["name"]
+                        names = old_config_map.get(kind, [])
+                        if name in names:
+                            names.remove(name)
+
+                        if kind == "Service":
+                            pod_name = name
+                        res = self.read(kind, name, namespace)
+                        if hasattr(res, "status") and res.status == 404 and conf:
+                            self.create(kind, name, namespace, conf)
+                        elif res.metadata:
```

Comment:
There is a corner case. If the `kube.read` fails in other ways than `404` not found, it will execute line 348 and it will fail because `res` will be an `APIException`